web-common/src/features/dashboards/url-state/getBasePreset.ts

```diff
@@ -0,0 +1,51 @@
+import { createAndExpression } from "@rilldata/web-common/features/dashboards/stores/filter-utils";
+import {
+  URLStateDefaultTDDChartType,
+  URLStateDefaultTimeRange,
+  URLStateDefaultTimezone,
+} from "@rilldata/web-common/features/dashboards/url-state/defaults";
+import { ToURLParamTDDChartMap } from "@rilldata/web-common/features/dashboards/url-state/mappers";
+import type { LocalUserPreferences } from "@rilldata/web-common/features/dashboards/user-preferences";
+import {
+  V1ExploreComparisonMode,
+  type V1ExplorePreset,
+  type V1ExploreSpec,
+  V1ExploreWebView,
+} from "@rilldata/web-common/runtime-client";
+
+export function getBasePreset(
+  explore: V1ExploreSpec,
+  preferences: LocalUserPreferences,
+) {
```

Comment:
It feels like this function is doing two things: 1. defining the "base" ("default"?) state for all Explores and 2. merging the generic default state with the given Explore's default state and the user preferences.

Could we define a `DEFAULT_EXPLORE_STATE` in a constant outside this function? And maybe the function name should be reconsidered.